### PR TITLE
Generate random password

### LINKED ISSUE #4931: Proposal: gitea admin create-user random password
- Gitea version (or commit ref): 1.6.x
- Git version: any
- Operating system: any
- Database (use `[x]`): any

## Description

It would be good addition to #4489 (Force user to change password), as new admin-created users have temporary passwords - would be nice to add random password generator option to ```gitea admin create-user``` commans.

Example syntax: ```gitea admin create-user --name=jane --email=jane@example.org --random-password```

Or just imply ```--random-password``` if no ```--password=xxxxxxxx``` set.

If random password parameter specified, new password should be printed to stdout on successful user creation to make admin copy it and send to new user. Or if mail is configured, this temporary password may be also sent to new user's email.

Probably random password should be composed of characters which are hard to mistype (without 0 O, i l 1 and other similar pairs)

### PR DESCRIPTION
**Pull request by adelowo**
Fixes https://github.com/go-gitea/gitea/issues/4931

- [x] Added a new cli flag `-random-password`
- [x] Added a new cli flag `-random-password-length` for configuring length of generated password
- [x]  Generate a random password with `generate.GetRandomString`

Thoughts ?

### CODE REVIEW

**Comment by lafriks** [comment]
If implemented sending email should probably be optional with special flag

**Comment by adelowo** [comment]
> If implemented sending email should probably be optional with special flag

Will add that

**Comment by adelowo** [comment]
@lafriks I couldn't get emails to work... Would work on that and send another PR in the future

File: cmd/admin.go [hunk 1]
```diff
@@ -277,10 +283,29 @@ func runChangePassword(c *cli.Context) error {
...
+	if c.IsSet("password") && c.IsSet("random-password") {
+		return errors.New("cannot set both -random-password and -password flags")
+	}
+
+	var password string
+
+	if c.IsSet("password") {
+		password = c.String("password")
+	} else if c.IsSet("random-password") {
+		password, err := generate.GetRandomString(30)
```
Comment (lafriks) [inline cmd/admin.go]: is it really reasonable to generate 30 characters long password?

Comment (lunny) [reply cmd/admin.go]: Maybe give another command line argument `password length`.

Comment (adelowo) [reply cmd/admin.go]: @lafriks my passwords are usually 40 - 75 characters long ( including characters ), I probably added that based on what I'd use. 😃😃😃

@lunny nice enough? What do you think @lafriks?

Comment (lafriks) [reply cmd/admin.go]: Option to specify length would be great. Default should probably be something like 12

**Review by sapk — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**